R/importConservation.R

```diff
@@ -0,0 +1,164 @@
+# .importConservation.R
+
+#' Generate a data file containing conservation score and chromosome position
+#' pairs.
+#'
+#' \code{.importConservation} imports conservation scores from a UCSC chromosome
+#' conservation score file. Generate a big matrix (col size 2, row size equal to
+#' the number of rows of the input file) containing conservation score and
+#' chromosome position pairs.
+#'
+#' @param fName The path to the UCSC chromosome conservation score file.
+#' @param outName The path to an output file for the rCNA RDS object.
+#' @param scoringType The type of the conservation score.
+#' @param silent Controls whether writing the result to the global logfile is
+#'   enabled. TRUE by default.
+#' @param writeLog  Return rCNA object of COSMIC CNA data as RDS compressed data frame file.
+
+.importConservation <- function(fName,
+                                outName,
+                                scoringType,
+                                silent = FALSE,
+                                writeLog = TRUE) {
+
+    # ==== PARAMETER TYPE VALIDATION ==========================================
+
+    # General parameter checks
+    cR <- character()
+    cR <- c(cR, .checkArgs(fName,     like = "FILE_E",   checkSize = TRUE))
+    cR <- c(cR, .checkArgs(outName,     like = "a",        checkSize = TRUE))
+    cR <- c(cR, .checkArgs(scoringType,     like = "a",        checkSize = TRUE))
+    cR <- c(cR, .checkArgs(silent,       like = logical(1), checkSize = TRUE))
+    cR <- c(cR, .checkArgs(writeLog,     like = logical(1), checkSize = TRUE))
+
+    if (length(cR) > 0) {
+        stop(cR)
+    }
+
+    # ==== READ UCSC Conservation DATA ========================================
+
+    if (!silent) {
+        cat("Reading UCSC Conservation data from ", fName, " ...\n")
+    }
+
+    conservationFile <- file(fName, open = "r")
+
+    # Read header file
+    tmp <- readLines(conservationFile, n = 1)
+    header <- unlist(strsplit(tmp, " "))
+
+    chrCol <- grep("chr", header)
+    startCol <- grep("start", header)
+
+    if (length(chrCol) == 0 || length(startCol) == 0) {
+        errorMessage <- "Invalid header."
+        stop(errorMessage)
+    }
+
+    chromosomeSource <- unlist(strsplit(header[chrCol], "="))[2]
+    currPos <- as.numeric(unlist(strsplit(header[startCol], "="))[2])
+    currIndex <- 1
+
+
+    # Determine the number of lines
+    lineCountFile <- file(fName, open = "r")
```

Comment:
You are reading the entire GB-scale file to determine the number of records? And reading chunks? And calling them "numLines"? 

How about:
```
system(paste("wc -l", fName))
#... and parsing out the number...
as.numeric(strsplit(system(paste("wc -l", fName), intern = TRUE), " +")[[1]][2])
```